src/html-media-source.js

```diff
@@ -264,4 +264,41 @@ export default class HtmlMediaSource extends videojs.EventTarget {
     this.sourceBuffers.push(buffer);
     return buffer;
   }
+
+  /**
+   * add a property to the VSB indicating the current audio/video/text stream
+   * properties (sample rate, channels, etc) and fire an event
+   * when the info changes
+   *
+   * @private
+   * @param {String} type the type of stream to change info on (audio, video, text)
+   * @param {Object} info info about the currently muxed audio/video/text stream
+   */
+  changeInfo_(type, info) {
+    let propName = `${type}Info_`;
+    let old = {};
+
+    if (!this[propName]) {
```

Comment:
Should we also check that info and this[propName] have the same number of elements? If info is empty, it looks like everything would be fine (unless we expect that behavior).